### ISSUE TITLE
[HOLD #48778]Report action not properly updating when submitting expense on behalf of someone

### ISSUE THREAD

**Issue by m-natarajan** [Expensify/App#49069]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

**Version Number:** 9.0.33-1
**Reproducible in staging?:** Y
**Reproducible in production?:** Y
**If this was caught during regression testing, add the test name, ID and link from TestRail:**
**Email or phone of affected tester (no customers):**
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Expensify/Expensify Issue URL:**
**Issue reported by:** @yuwenmemon 
**Slack conversation:** https://expensify.slack.com/archives/C049HHMV9SM/p1726082993070849

## Action Performed:
1. Submitter draft an expense in workspace
2. Approver (who is NOT an ADMIN) submits the expense on behalf of submitter
## Expected Result:
Report action as` submitted $XX.XX (on behalf of XXX)` or indicate that somehow approver took control of the report to take the action
## Actual Result:
Report action says submitted $XX.XX 
## Workaround:
Unknown
## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [ ] Android: Native
- [ ] Android: mWeb Chrome
- [ ] iOS: Native
- [ ] iOS: mWeb Safari
- [x] MacOS: Chrome / Safari
- [ ] MacOS: Desktop

## Screenshots/Videos

![Screenshot 2024-09-11 at 12 27 35 PM](https://github.com/user-attachments/assets/ebbc0f2b-546d-4839-897f-a62a741de04b)

![Snip - (1) New Expensify - Google Chrome (2)](https://github.com/user-attachments/assets/53a874ea-1e09-4445-bb03-8ec18d1923ce)

Add any screenshot/video evidence

</details>

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

**Comment by ChavdaSachin** [comment]
https://github.com/Expensify/App/issues/48778

**Comment by ChavdaSachin** [comment]
https://github.com/Expensify/App/issues/48778#issue-2513522701

**Comment by ChavdaSachin** [comment]
Perhaps Approver should not be able to submit report on behalf of submitter.
https://github.com/Expensify/App/issues/48778

**Comment by slafortune** [comment]
@ChavdaSachin I believe your correct, we should put this on hold for **Non-admin approver should be able to Submit draft reports on behalf of their submitters #48778**

<img width="671" alt="image" src="https://github.com/user-attachments/assets/ee153610-3a9d-40e6-a87a-d9cea344a1ba">

Update original issue

**Comment by slafortune** [comment]
Still on hold

**Comment by slafortune** [comment]
Still on hold

**Comment by ChavdaSachin** [comment]
## Proposal




### Please re-state the problem that we are trying to solve in this issue.
Report action not properly updating when submitting expense on behalf of someone.


### What is the root cause of that problem?
Currently We do not store information about who submitted the report.
So without storing `submittedByAccountID` it would not be possible to display the message we want here.

### What changes do you think we should make in order to solve the problem?
Include a new field `submittedByAccountID` to report, store it when user submits the report, and later make use of `submittedByAccountID` to display avatar , name and message - "submitted $XX.XX (on behalf of XXX)"

We might need BE support here.

### What alternative solutions did you explore? (Optional)




**Reminder:** Please use plain English, be brief and avoid jargon. Feel free to use images, charts or pseudo-code if necessary. Do not post large multi-line diffs or write walls of text. Do not create PRs unless you have been hired for this job.

**Comment by slafortune** [comment]
@ChavdaSachin this is on hold for a bigger project -https://github.com/Expensify/App/issues/49069#issuecomment-2358968829
We'll retest to see if it's still an issue once that's complete.

**Comment by slafortune** [comment]
Not overdue - updated to weekly, waiting on a retest

**Comment by garrettmknight** [comment]
Taking this one over.

**Comment by garrettmknight** [comment]
Tested, fixed, closing.
